hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -252,6 +254,9 @@ protected Option<Pair<T, Map<String, Object>>> doProcessNextDataRecord(T record,
       // NOTE: Record have to be cloned here to make sure if it holds low-level engine-specific
       //       payload pointing into a shared, mutable (underlying) buffer we get a clean copy of
       //       it since these records will be put into records(Map).
+      Comparable orderingValue = readerContext.getOrderingValue(
+          Option.of(record), metadata, readerSchema, orderingFieldName, orderingFieldType, orderingFieldDefault);
+      metadata.put(INTERNAL_META_ORDERING_FIELD, orderingValue);
```

Comment:
why were we not doing this before? I am a little uneasy about this.

Reply:
In readerContext.getOrderingValue we are setting it now. Maybe this wasn't the case when you started the pr

Reply:
I don't think this is needed

Reply:
ok, let me remove it and see how it works.

Reply:
tests passed. Done.